portal/src/model/group.ts

```diff
@@ -0,0 +1,39 @@
+import { Group } from "../graphql/adminapi/globalTypes.generated";
+
+export interface SearchableGroup extends Pick<Group, "id" | "key" | "name"> {}
+
+export function searchGroups<G extends SearchableGroup>(
+  groups: G[],
+  searchKeyword: string
+): G[] {
+  if (searchKeyword === "") {
+    return groups;
+  }
+  const keywords = searchKeyword
+    .toLowerCase()
+    .split(" ")
+    .flatMap((keyword) => {
+      const trimmedKeyword = keyword.trim();
+      if (trimmedKeyword) {
+        return [trimmedKeyword];
+      }
+      return [];
+    });
+  return groups.filter((group) => {
+    const groupID = group.id.toLowerCase();
+    const groupKey = group.key.toLowerCase();
+    const groupName = group.name?.toLowerCase();
+    for (const keyword of keywords) {
+      if (groupID === keyword) {
+        return true;
+      }
+      if (groupKey.includes(keyword)) {
+        return true;
+      }
+      if (groupName?.includes(keyword)) {
+        return true;
+      }
+    }
+    return false;
+  });
+}
```

Comment:
We do client side filtering here?

Reply:
Because I doesn't find a api which search groups inside roles. We can confirm in standup.